rfcs/text/0006_management_section_service.md

```diff
@@ -0,0 +1,343 @@
+    kibanaSection.registerLink({ id: 'my-kibana-section-link', ... });
+  }
+
+  start(core, { management }) {
+    // access all registered sections, filtered based on capabilities
+    management.sections.getAvailable();
+    // automatically navigate to any section link by id
+    management.sections.navigateToSectionLink('my-kibana-section-link');
+  }
+}
+ 
+// my_plugin/public/my-section.tsx
+
+export function renderApp(context, { sectionBasePath, element }) {
+  ReactDOM.render(
+    // `sectionBasePath` would be `/app/management/my-section/my-link`
+    <MyApp basename={sectionBasePath} />,
+    element
+  );
+ 
+  // return value must be a function that unmounts (just like Core Application Service)
+  return () => ReactDOM.unmountComponentAtNode(element);
+}
+```
+
+We can also create a utility in `kibana_react` to make it easy for folks to `mount` a React app:
+```ts
+// src/plugins/kibana_react/public/mount_with_react.tsx
+import { KibanaContextProvider } from './context';
+
+export const mountWithReact = (
+  Component: React.ComponentType<{ basename: string }>,
+  context: AppMountContext,
+  params: ManagementSectionMountParams,
+) => {
+  ReactDOM.render(
+    (
+      <KibanaContextProvider services={{ ...context }}>
+        <Component basename={params.sectionBasePath} />
+      </KibanaContextProvider>
+    ),
+    params.element
+  );
+
+  return () => ReactDOM.unmountComponentAtNode(params.element);
+}
+
+// my_plugin/public/plugin.ts
+import { mountWithReact } from 'src/plugins/kibana_react/public';
+
+export class MyPlugin {
+  setup(core, { management }) {
+    const kibanaSection = management.sections.get('kibana');
+    kibanaSection.registerLink({
+      id: 'my-other-kibana-section-link',
+      ...,
+      async mount(context, params) {
+        const { MySection } = await import('./components/my-section');
+        return mountWithReact(MySection, context, params);
+      }
+    });
+  }
+}
+```
+ 
+# Detailed design
+
+```ts
+interface ManagementSetup {
+  sections: SectionsServiceSetup;
+}
+
+interface ManagementStart {
+  sections: SectionsServiceStart;
+}
+ 
+interface SectionsServiceSetup {
+  get: (sectionId: string) => Section;
+  getAvailable: () => Section[]; // filtered based on capabilities
+  register: Register;
+}
+
+interface SectionsServiceStart {
+  getAvailable: () => Array<Omit<Section, 'registerLink'>>; // filtered based on capabilities
+  // uses `core.application.navigateToApp` under the hood, automatically prepending the `path` for the link
+  navigateToSectionLink: (linkId: string, options?: { path?: string; state?: any }) => void;
+}
+
+type Register = (
+  id: string,
+  title: string,
+  description: string, // not used in current UI, but possibly in future
+  order?: number,
+  euiIconType?: string, // takes precedence over `icon` property.
+  icon?: string, // URL to image file; fallback if no `euiIconType`
+) => Section | Error;
```

Comment:
Would this return an Error object? I'd expect it to throw instead of return. Same of the `RegisterLink` type below.

Reply:
Ah, nope this was a typo on my part. The intent would be for both to throw an error; will update